### DISCUSSION TITLE
ProgramTransformer generate wrong source map

### DISCUSSION THREAD

**Discussion by corrm** [nonara/ts-patch#80] [Q&A]
Hello,

i use same example here https://github.com/nonara/ts-patch/discussions/29
generated JS is correct and as expected BUT

for source files that transformer doesn't touch will just get a wrong source map as here [source-map-visualization](http://sokra.github.io/source-map-visualization/#base64,aW1wb3J0IHsgX19kZWNvcmF0ZSB9IGZyb20gInRzbGliIjsNCmltcG9ydCB7IEZhbmN5TWVzc2FnZVR5cGUgfSBmcm9tICIuL01lc3NhZ2UiOw0KaW1wb3J0IHsgQ29tcG9uZW50U3luY01lc3NhZ2UgfSBmcm9tICIuL01lc3NhZ2UvQ29tcG9uZW50cyI7DQovKioNCiAqIEp1c3QgYSBkZWNvcmF0b3IgbWFya2VyDQogKi8NCmZ1bmN0aW9uIG1lc3NhZ2VIYW5kbGVyKG1lc3NhZ2VUeXBlKSB7DQogICAgcmV0dXJuICh0YXJnZXQsIHByb3BlcnR5S2V5LCBkZXNjcmlwdG9yKSA9PiB7DQogICAgfTsNCn0NCmV4cG9ydCBjbGFzcyBGYW5jeU1lc3NhZ2VIYW5kbGVyIHsNCiAgICBzdGF0aWMgY3VzdG9tRGF0YUhhbmRsZXIoY2xpZW50LCBtZXNzYWdlKSB7DQogICAgfQ0KICAgIHN0YXRpYyBhc3luYyBzZXJ2ZXJSZWFkeUhhbmRsZXIoY2xpZW50LCBtZXNzYWdlKSB7DQogICAgICAgIGNsaWVudC5vblNlcnZlclJlYWR5Lm5leHQobWVzc2FnZS5zZXJ2ZXJSZWFkeSk7DQogICAgICAgIGNvbnN0IG1zZ1RvU2VuZCA9IG5ldyBDb21wb25lbnRTeW5jTWVzc2FnZSgpOw0KICAgICAgICBjbGllbnQuc2VuZChGYW5jeU1lc3NhZ2VUeXBlLkNPTVBPTkVOVF9TWU5DLCBtc2dUb1NlbmQpOw0KICAgIH0NCiAgICBzdGF0aWMgY29tcG9uZW50U3luY0hhbmRsZXIoY2xpZW50LCBtZXNzYWdlKSB7DQogICAgICAgIGNvbnN0IGNvbXBvbmVudHMgPSBtZXNzYWdlLmNvbXBvbmVudHNJZDsNCiAgICB9DQogICAgLyoqDQogICAgICogSGFuZGxlIG1lc3NhZ2VzIHRoYXQgZ2V0IGZyb20gc2VydmVyDQogICAgICoNCiAgICAgKiBAc3RhdGljDQogICAgICogQHBhcmFtIHtGYW5jeUNsaWVudH0gY2xpZW50IENsaWVudCBpbnN0YW5jZSB0aGF0IHJlY2VpdmVkIHRoZSBtZXNzYWdlDQogICAgICogQHBhcmFtIHtGYW5jeU1lc3NhZ2VUeXBlfSBtc2dUeXBlIE1lc3NhZ2UgdHlwZQ0KICAgICAqIEBwYXJhbSB7VWludDhBcnJheX0gbXNnQnl0ZXMgbWVzc2FnZSBieXRlcw0KICAgICAqDQogICAgICogQG1lbWJlck9mIEZhbmN5TWVzc2FnZUhhbmRsZXINCiAgICAgKi8NCiAgICBzdGF0aWMgYXN5bmMgaGFuZGxlTWVzc2FnZUFzeW5jKGNsaWVudCwgbXNnVHlwZSwgbXNnQnl0ZXMpIHsNCiAgICAgICAgLy8gUHJvdG9idWZUcmFuc2Zvcm1lciB3aWxsIGZpbGwgdGhpcyBtZXRob2QgYm9keQ0KICAgIH0NCn0NCl9fZGVjb3JhdGUoWw0KICAgIG1lc3NhZ2VIYW5kbGVyKEZhbmN5TWVzc2FnZVR5cGUuQ1VTVE9NX0RBVEEpDQpdLCBGYW5jeU1lc3NhZ2VIYW5kbGVyLCAiY3VzdG9tRGF0YUhhbmRsZXIiLCBudWxsKTsNCl9fZGVjb3JhdGUoWw0KICAgIG1lc3NhZ2VIYW5kbGVyKEZhbmN5TWVzc2FnZVR5cGUuU0VSVkVSX1JFQURZKQ0KXSwgRmFuY3lNZXNzYWdlSGFuZGxlciwgInNlcnZlclJlYWR5SGFuZGxlciIsIG51bGwpOw0KX19kZWNvcmF0ZShbDQogICAgbWVzc2FnZUhhbmRsZXIoRmFuY3lNZXNzYWdlVHlwZS5DT01QT05FTlRfU1lOQykNCl0sIEZhbmN5TWVzc2FnZUhhbmRsZXIsICJjb21wb25lbnRTeW5jSGFuZGxlciIsIG51bGwpOw0KLy8jIHNvdXJjZU1hcHBpbmdVUkw9ZmFuY3ktbWVzc2FnZS1oYW5kbGVyLmpzLm1hcA==,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiZmFuY3ktbWVzc2FnZS1oYW5kbGVyLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiLi4vc3JjL2ZhbmN5LW1lc3NhZ2UtaGFuZGxlci50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiO0FBQ0EsT0FBTyxFQUFFLGdCQUFnQixFQUFFLE1BQU0sV0FBVyxDQUFDO0FBQzdDLE9BQU8sRUFBRSxvQkFBb0IsRUFBRSxNQUFNLHNCQUFzQixDQUFDO0FBRTVEOztHQUVHO0FBQ0gsU0FBUyxjQUFjLENBQUMsV0FBNkI7SUFDakQsT0FBTyxDQUFDLE1BQVcsRUFBRSxXQUE0QixFQUFFLFVBQThCLEVBQUUsRUFBRTtJQUNyRixDQUFDLENBQUM7QUFDTixDQUFDO0FBQ0QsTUFBTSxPQUFPLG1CQUFtQjtJQUVwQixNQUFNLENBQUMsaUJBQWlCLENBQUMsTUFBbUIsRUFBRSxPQUEwQjtJQUNoRixDQUFDO0lBRU8sTUFBTSxDQUFDLEtBQUssQ0FBQyxrQkFBa0IsQ0FBQyxNQUFtQixFQUFFLE9BQTJCO1FBQ3BGLE1BQU0sQ0FBQyxhQUFhLENBQUMsSUFBSSxDQUFDLE9BQU8sQ0FBQyxXQUFXLENBQUMsQ0FBQztRQUMvQyxNQUFNLFNBQVMsR0FBRyxJQUFJLG9CQUFvQixFQUFFLENBQUM7UUFDN0MsTUFBTSxDQUFDLElBQUksQ0FBQyxnQkFBZ0IsQ0FBQyxjQUFjLEVBQUUsU0FBUyxDQUFDLENBQUM7SUFDNUQsQ0FBQztJQUVPLE1BQU0sQ0FBQyxvQkFBb0IsQ0FBQyxNQUFtQixFQUFFLE9BQTZCO1FBQ2xGLE1BQU0sVUFBVSxHQUFhLE9BQU8sQ0FBQyxZQUFZLENBQUM7SUFDdEQsQ0FBQztJQUNEOzs7Ozs7Ozs7T0FTRztJQUNJLE1BQU0sQ0FBQyxLQUFLLENBQUMsa0JBQWtCLENBQUMsTUFBbUIsRUFBRSxPQUF5QixFQUFFLFFBQW9CO1FBQ3ZHLGlEQUFpRDtJQUNyRCxDQUFDO0NBQ0o7QUF6Qkc7SUFEQyxjQUFjLENBQUMsZ0JBQWdCLENBQUMsV0FBVyxDQUFDO2tEQUU1QztBQUVEO0lBREMsY0FBYyxDQUFDLGdCQUFnQixDQUFDLFlBQVksQ0FBQzttREFLN0M7QUFFRDtJQURDLGNBQWMsQ0FBQyxnQkFBZ0IsQ0FBQyxjQUFjLENBQUM7cURBRy9DIn0=,aW1wb3J0IHsgRmFuY3lDbGllbnQgfSBmcm9tICIuL2ZhbmN5LWNsaWVudCI7DQppbXBvcnQgeyBGYW5jeU1lc3NhZ2VUeXBlIH0gZnJvbSAiLi9NZXNzYWdlIjsNCmltcG9ydCB7IENvbXBvbmVudFN5bmNNZXNzYWdlIH0gZnJvbSAiLi9NZXNzYWdlL0NvbXBvbmVudHMiOw0KaW1wb3J0IHsgQ3VzdG9tRGF0YU1lc3NhZ2UsIFNlcnZlclJlYWR5TWVzc2FnZSB9IGZyb20gIi4vTWVzc2FnZS9Db3JlIjsNCg0KLyoqDQogKiBKdXN0IGEgZGVjb3JhdG9yIG1hcmtlcg0KICovDQpmdW5jdGlvbiBtZXNzYWdlSGFuZGxlcihtZXNzYWdlVHlwZTogRmFuY3lNZXNzYWdlVHlwZSk6IE1ldGhvZERlY29yYXRvcg0Kew0KICAgIHJldHVybiAodGFyZ2V0OiBhbnksIHByb3BlcnR5S2V5OiBzdHJpbmcgfCBzeW1ib2wsIGRlc2NyaXB0b3I6IFByb3BlcnR5RGVzY3JpcHRvcikgPT4NCiAgICB7DQogICAgfTsNCn0NCg0KZXhwb3J0IGNsYXNzIEZhbmN5TWVzc2FnZUhhbmRsZXINCnsNCiAgICBAbWVzc2FnZUhhbmRsZXIoRmFuY3lNZXNzYWdlVHlwZS5DVVNUT01fREFUQSkNCiAgICBwcml2YXRlIHN0YXRpYyBjdXN0b21EYXRhSGFuZGxlcihjbGllbnQ6IEZhbmN5Q2xpZW50LCBtZXNzYWdlOiBDdXN0b21EYXRhTWVzc2FnZSk6IHZvaWQNCiAgICB7DQogICAgfQ0KDQogICAgQG1lc3NhZ2VIYW5kbGVyKEZhbmN5TWVzc2FnZVR5cGUuU0VSVkVSX1JFQURZKQ0KICAgIHByaXZhdGUgc3RhdGljIGFzeW5jIHNlcnZlclJlYWR5SGFuZGxlcihjbGllbnQ6IEZhbmN5Q2xpZW50LCBtZXNzYWdlOiBTZXJ2ZXJSZWFkeU1lc3NhZ2UpOiBQcm9taXNlPHZvaWQ+DQogICAgew0KICAgICAgICBjbGllbnQub25TZXJ2ZXJSZWFkeS5uZXh0KG1lc3NhZ2Uuc2VydmVyUmVhZHkpOw0KDQogICAgICAgIGNvbnN0IG1zZ1RvU2VuZCA9IG5ldyBDb21wb25lbnRTeW5jTWVzc2FnZSgpOw0KICAgICAgICBjbGllbnQuc2VuZChGYW5jeU1lc3NhZ2VUeXBlLkNPTVBPTkVOVF9TWU5DLCBtc2dUb1NlbmQpOw0KICAgIH0NCg0KICAgIEBtZXNzYWdlSGFuZGxlcihGYW5jeU1lc3NhZ2VUeXBlLkNPTVBPTkVOVF9TWU5DKQ0KICAgIHByaXZhdGUgc3RhdGljIGNvbXBvbmVudFN5bmNIYW5kbGVyKGNsaWVudDogRmFuY3lDbGllbnQsIG1lc3NhZ2U6IENvbXBvbmVudFN5bmNNZXNzYWdlKTogdm9pZA0KICAgIHsNCiAgICAgICAgY29uc3QgY29tcG9uZW50czogc3RyaW5nW10gPSBtZXNzYWdlLmNvbXBvbmVudHNJZDsNCiAgICB9DQoNCiAgICAvKioNCiAgICAgKiBIYW5kbGUgbWVzc2FnZXMgdGhhdCBnZXQgZnJvbSBzZXJ2ZXINCiAgICAgKg0KICAgICAqIEBzdGF0aWMNCiAgICAgKiBAcGFyYW0ge0ZhbmN5Q2xpZW50fSBjbGllbnQgQ2xpZW50IGluc3RhbmNlIHRoYXQgcmVjZWl2ZWQgdGhlIG1lc3NhZ2UNCiAgICAgKiBAcGFyYW0ge0ZhbmN5TWVzc2FnZVR5cGV9IG1zZ1R5cGUgTWVzc2FnZSB0eXBlDQogICAgICogQHBhcmFtIHtVaW50OEFycmF5fSBtc2dCeXRlcyBtZXNzYWdlIGJ5dGVzDQogICAgICoNCiAgICAgKiBAbWVtYmVyT2YgRmFuY3lNZXNzYWdlSGFuZGxlcg0KICAgICAqLw0KICAgIHB1YmxpYyBzdGF0aWMgYXN5bmMgaGFuZGxlTWVzc2FnZUFzeW5jKGNsaWVudDogRmFuY3lDbGllbnQsIG1zZ1R5cGU6IEZhbmN5TWVzc2FnZVR5cGUsIG1zZ0J5dGVzOiBVaW50OEFycmF5KTogUHJvbWlzZTx2b2lkPg0KICAgIHsNCiAgICAgICAgLy8gUHJvdG9idWZUcmFuc2Zvcm1lciB3aWxsIGZpbGwgdGhpcyBtZXRob2QgYm9keQ0KICAgIH0NCn0NCg==)
for source file that transformer edit will generate corrupted source map file
![image](https://user-images.githubusercontent.com/5868825/183108794-396f24f4-393a-4f59-a337-d4baf4cf2ba1.png)

i read that https://github.com/nonara/ts-patch/discussions/29#discussioncomment-971127
but there is any solution ?

**Comment by nonara** [comment]
Hello. Thanks for bringing this to my attention.

Sorry for the delay on the response. I can see a SourceMap not properly lining up if you transform it with a Program transformer. That will be an issue if you transform using a Program transformer. A _corrupt_ sourcemap is unexpected. I'd have to get a closer look.

Can you open this as an issue, ideally with a small repro? This would help a lot toward being able to check out the issue.
